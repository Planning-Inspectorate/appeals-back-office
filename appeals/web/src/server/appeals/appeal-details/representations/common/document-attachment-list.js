/** @typedef {import("#appeals/appeal-details/representations/types.js").Representation} Representation */

import { buildHtmUnorderedList } from '#lib/nunjucks-template-builders/tag-builders.js';
import { mapDocumentDownloadUrl } from '#appeals/appeal-documents/appeal-documents.mapper.js';

export const getAttachmentList = (/** @type {Representation} */ representation) => {
	const filteredAttachments = representation.attachments?.filter((attachment) => {
		const { isDeleted, latestVersionId } = attachment?.documentVersion?.document ?? {};
		return latestVersionId === attachment.version && !isDeleted;
	});

	return filteredAttachments.length > 0
		? buildHtmUnorderedList(
				filteredAttachments.map(
					(a) =>
						`<a class="govuk-link" href="${mapDocumentDownloadUrl(
							a.documentVersion.document.caseId,
							a.documentVersion.document.guid,
							a.documentVersion.document.name
						)}" target="_blank">${a.documentVersion.document.name}</a>`
				)
		  )
		: null;
};
