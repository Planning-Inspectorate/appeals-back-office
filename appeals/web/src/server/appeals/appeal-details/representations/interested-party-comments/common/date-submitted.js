import { applyEdits } from '#lib/edit-utilities.js';
import { dateSubmitted } from '../../document-attachments/add-document.mapper.js';

/** @typedef {import("../../../appeal-details.types.js").WebAppeal} Appeal */
/** @typedef {import('#appeals/appeal-details/representations/types.js').Representation} Representation */
/** @typedef {{ 'day': string, 'month': string, 'year': string }} RequestDate */
/** @typedef {RequestDate} ReqBody */

/**
 * @param {object} options
 * @param {(appealDetails: Appeal, comment: Representation) => string} options.getBackLinkUrl
 * @param {(request: import('@pins/express').Request) => RequestDate} options.getValue
 * @returns {import('@pins/express').RenderHandler<{}, {}, ReqBody>}
 */
export const renderDateSubmittedFactory =
	({ getBackLinkUrl, getValue }) =>
	(request, response) => {
		const backLinkUrl = getBackLinkUrl(request.currentAppeal, request.currentRepresentation);
		const value = getValue(request);

		const pageContent = dateSubmitted(
			request.currentAppeal,
			request.errors,
			value,
			backLinkUrl,
			{}
		);

		return response.status(request.errors ? 400 : 200).render('patterns/change-page.pattern.njk', {
			errors: request.errors,
			pageContent
		});
	};

/**
 * @param {object} options
 * @param {(appealDetails: Appeal, comment: Representation) => string} options.getRedirectUrl
 * @param {import('@pins/express').RenderHandler<{}, {}, ReqBody>} options.errorHandler
 * @returns {import('@pins/express').RenderHandler<{}, {}, ReqBody>}
 */
export const postDateSubmittedFactory =
	({ getRedirectUrl, errorHandler }) =>
	(request, response, next) => {
		if (request.errors) {
			return errorHandler(request, response, next);
		}

		applyEdits(request, 'addIpComment');

		const { currentAppeal, currentRepresentation } = request;

		return response.redirect(getRedirectUrl(currentAppeal, currentRepresentation));
	};
