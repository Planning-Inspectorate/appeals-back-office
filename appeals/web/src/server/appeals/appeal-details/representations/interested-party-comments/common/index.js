export { postDateSubmittedFactory, renderDateSubmittedFactory } from './date-submitted.js';
export { postRedactionStatusFactory, renderRedactionStatusFactory } from './redaction-status.js';
