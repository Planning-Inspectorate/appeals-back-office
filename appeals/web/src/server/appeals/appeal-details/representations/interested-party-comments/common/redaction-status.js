import { appealShortReference } from '#lib/appeals-formatter.js';
import { radiosInput } from '#lib/mappers/index.js';
import { APPEAL_REDACTED_STATUS } from '@planning-inspectorate/data-model';

/** @typedef {import("../../../appeal-details.types.js").WebAppeal} Appeal */
/** @typedef {import('#appeals/appeal-details/representations/types.js').Representation} Representation */
/** @typedef {{ 'redactionStatus': string }} ReqBody */

export const statusFormatMap = {
	[APPEAL_REDACTED_STATUS.REDACTED]: 'Redacted',
	[APPEAL_REDACTED_STATUS.NOT_REDACTED]: 'Unredacted',
	[APPEAL_REDACTED_STATUS.NO_REDACTION_REQUIRED]: 'No redaction required'
};

export const name = 'redactionStatus';

/**
 * @param {any} maybeRedactionStatus
 * @returns {maybeRedactionStatus is keyof APPEAL_REDACTED_STATUS}
 */
export const isValidRedactionStatus = (maybeRedactionStatus) =>
	Object.keys(statusFormatMap).includes(maybeRedactionStatus);
/**
 * @param {Appeal} appealDetails
 * @param {import('@pins/express').ValidationErrors | undefined} errors
 * @param {string} value
 * @param {string} backLinkUrl
 * @returns {PageContent}
 */
const mapper = (appealDetails, errors, value, backLinkUrl) => ({
	title: 'Redaction status',
	backLinkUrl,
	preHeading: `Appeal ${appealShortReference(appealDetails.appealReference)}`,
	pageComponents: [
		radiosInput({
			name,
			legendText: 'Redaction status',
			legendIsPageHeading: true,
			items: Object.entries(statusFormatMap).map(([value, text]) => ({ value, text })),
			value
		})
	]
});

/**
 * @param {object} options
 * @param {(appealDetails: Appeal, comment: Representation) => string} options.getBackLinkUrl,
 * @param {(request: import('@pins/express').Request) => string} options.getValue
 * @returns {import('@pins/express').RenderHandler<{}, {}, ReqBody>}
 */
export const renderRedactionStatusFactory =
	({ getBackLinkUrl, getValue }) =>
	(request, response) => {
		const backLinkUrl = getBackLinkUrl(request.currentAppeal, request.currentRepresentation);
		const value = getValue(request);

		const pageContent = mapper(request.currentAppeal, request.errors, value, backLinkUrl);

		return response.status(request.errors ? 400 : 200).render('patterns/change-page.pattern.njk', {
			errors: request.errors,
			pageContent
		});
	};

/**
 * @param {object} options
 * @param {(appealDetails: Appeal, comment: Representation) => string} options.getRedirectUrl
 * @param {import('@pins/express').RenderHandler<{}, {}, ReqBody>} options.errorHandler
 * @returns {import('@pins/express').RenderHandler<{}, {}, ReqBody>}
 */
export const postRedactionStatusFactory =
	({ getRedirectUrl, errorHandler }) =>
	(request, response, next) => {
		if (request.errors) {
			return errorHandler(request, response, next);
		}

		const { currentAppeal, currentRepresentation } = request;

		return response.redirect(getRedirectUrl(currentAppeal, currentRepresentation));
	};
