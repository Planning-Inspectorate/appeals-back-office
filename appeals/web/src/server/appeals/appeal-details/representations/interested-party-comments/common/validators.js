import { createValidator } from '@pins/express';
import { createTextInputOptionalValidator } from '#lib/validators/text-input-validator.js';
import {
	createPostcodeValidator,
	createAddressLine1Validator,
	createAddressLine2Validator,
	createTownValidator
} from '#lib/validators/address.validator.js';
import { textInputCharacterLimits } from '#appeals/appeal.constants.js';

const maxLength = textInputCharacterLimits.defaultAddressInputLength;

export const validateInterestedPartyAddress = createValidator(
	createAddressLine1Validator(),
	createAddressLine2Validator(),
	createTownValidator(),
	createTextInputOptionalValidator(
		'county',
		maxLength,
		`County must be ${maxLength} characters or less`
	),
	createPostcodeValidator()
);
