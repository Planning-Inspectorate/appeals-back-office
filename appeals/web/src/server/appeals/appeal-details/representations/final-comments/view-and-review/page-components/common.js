import { buildHtmUnorderedList } from '#lib/nunjucks-template-builders/tag-builders.js';
import { preRenderPageComponents } from '#lib/nunjucks-template-builders/page-component-rendering.js';
import { APPEAL_REPRESENTATION_STATUS } from '@pins/appeals/constants/common.js';
import { mapDocumentDownloadUrl } from '#appeals/appeal-documents/appeal-documents.mapper.js';

/** @typedef {import('#appeals/appeal-details/representations/types.js').Representation} Representation */

/**
 * Maps representation types to their respective URL path fragments.
 * @param {string} representationType
 * @returns {string}
 */
function mapRepresentationTypeToPath(representationType) {
	switch (representationType) {
		case 'appellant_final_comment':
			return 'appellant';
		case 'lpa_final_comment':
			return 'lpa';
		default:
			throw new Error(`Unsupported representation type: ${representationType}`);
	}
}

/**
 * Generates the comment summary list used in both view and review pages.
 * @param {number} appealId
 * @param {Representation} comment - The comment object.
 * @returns {PageComponent} The generated comment summary list component.
 */
export function generateCommentsSummaryList(appealId, comment) {
	const commentIsDocument = !comment.originalRepresentation && comment.attachments?.length > 0;
	const folderId = comment.attachments?.[0]?.documentVersion?.document?.folderId ?? null;

	const filteredAttachments = comment.attachments?.filter((attachment) => {
		const { isDeleted, latestVersionId } = attachment?.documentVersion?.document ?? {};
		return latestVersionId === attachment.version && !isDeleted;
	});

	const attachmentsList = filteredAttachments?.length
		? buildHtmUnorderedList(
				filteredAttachments.map(
					(a) =>
						`<a class="govuk-link" href="${mapDocumentDownloadUrl(
							a.documentVersion.document.caseId,
							a.documentVersion.document.guid,
							a.documentVersion.document.name
						)}" target="_blank">${a.documentVersion.document.name}</a>`
				)
		  )
		: null;

	const commentTypePath = mapRepresentationTypeToPath(comment.representationType);

	const rows = [
		{
			key: { text: comment.redactedRepresentation ? 'Original final comments' : 'Final comments' },
			value: commentIsDocument
				? { text: 'Added as a document' }
				: {
						html: '',
						pageComponents: [
							{
								type: 'show-more',
								parameters: {
									text: comment.originalRepresentation,
									labelText: 'Read more'
								}
							}
						]
				  },
			actions: {
				items:
					comment.status === APPEAL_REPRESENTATION_STATUS.PUBLISHED
						? []
						: [
								{
									text: 'Redact',
									href: `/appeals-service/appeal-details/${appealId}/final-comments/${commentTypePath}/redact`,
									visuallyHiddenText: 'final comments'
								}
						  ]
			}
		},
		...(comment.redactedRepresentation
			? [
					{
						key: { text: 'Redacted comment' },
						value: {
							html: '',
							pageComponents: [
								{
									type: 'show-more',
									parameters: {
										text: comment.redactedRepresentation,
										labelText: 'Read more'
									}
								}
							]
						}
					}
			  ]
			: []),
		{
			key: { text: 'Supporting documents' },
			value: attachmentsList ? { html: attachmentsList } : { text: 'Not provided' },
			actions: {
				items: [
					...(filteredAttachments?.length > 0
						? [
								{
									text: 'Manage',
									href: `/appeals-service/appeal-details/${appealId}/final-comments/${commentTypePath}/manage-documents/${folderId}`,
									visuallyHiddenText: 'supporting documents'
								}
						  ]
						: []),
					{
						text: 'Add',
						href: `/appeals-service/appeal-details/${appealId}/final-comments/${commentTypePath}/add-document`,
						visuallyHiddenText: 'supporting documents'
					}
				]
			}
		}
	];

	/** @type {PageComponent} */
	const summaryList = {
		type: 'summary-list',
		wrapperHtml: {
			opening: '<div class="govuk-grid-row"><div class="govuk-grid-column-full">',
			closing: '</div></div>'
		},
		parameters: { rows }
	};

	preRenderPageComponents([summaryList]);

	return summaryList;
}
