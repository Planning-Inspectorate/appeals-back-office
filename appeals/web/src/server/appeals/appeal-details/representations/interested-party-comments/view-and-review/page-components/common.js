import { mapDocumentDownloadUrl } from '#appeals/appeal-documents/appeal-documents.mapper.js';
import { addressToMultilineStringHtml, representationHasAddress } from '#lib/address-formatter.js';
import { dateISOStringToDisplayDate } from '#lib/dates.js';
import { buildHtmlList } from '#lib/nunjucks-template-builders/tag-builders.js';
import { checkRedactedText } from '#lib/validators/redacted-text.validator.js';
import { COMMENT_STATUS } from '@pins/appeals/constants/common.js';

/** @typedef {import('#appeals/appeal-details/representations/types.js').Representation} Representation */

/**
 * @param {Representation} comment - The comment object.
 * @returns {string}
 */
const generateRejectionReasonsList = (comment) => {
	const listItemsString = comment.rejectionReasons
		.reduce(
			(listItems, { name, text }) =>
				// @ts-ignore
				text?.length
					? [...listItems, ...text.map((item) => name + ': ' + item)]
					: [...listItems, name],
			[]
		)
		// @ts-ignore
		.map((item) => `<li>${item}</li>`)
		.join('');
	return `<ul class="govuk-list govuk-list--bullet">${listItemsString}</ul>`;
};

/**
 * Generates the comment summary list used in both view and review pages.
 * @param {number} appealId
 * @param {Representation} comment - The comment object.
 * @param {{ isReviewPage?: boolean }} [options]
 * @returns {PageComponent} The generated comment summary list component.
 */
export function generateCommentSummaryList(
	appealId,
	comment,
	{ isReviewPage } = { isReviewPage: false }
) {
	const hasAddress = representationHasAddress(comment);
	const commentIsDocument = !comment.originalRepresentation && comment.attachments?.length;
	const folderId = comment.attachments?.[0]?.documentVersion?.document?.folderId ?? null;

	const filteredAttachments = comment.attachments?.filter((attachment) => {
		const { isDeleted, latestVersionId } = attachment?.documentVersion?.document ?? {};
		return latestVersionId === attachment.version && !isDeleted;
	});

	const attachmentsList = filteredAttachments?.length
		? buildHtmlList({
				items: filteredAttachments.map(
					(a) =>
						`<a class="govuk-link" href="${mapDocumentDownloadUrl(
							a.documentVersion.document.caseId,
							a.documentVersion.document.guid,
							a.documentVersion.document.name
						)}" target="_blank">${a.documentVersion.document.name}</a>`
				),
				isOrderedList: true,
				isNumberedList: filteredAttachments.length > 1
		  })
		: null;

	const { address, name, email } = comment.represented || {};
	const redactMatching = checkRedactedText(
		comment.originalRepresentation,
		comment.redactedRepresentation
	);
	const rows = [
		{
			key: { text: 'Interested party' },
			value: { text: name || 'Not provided' }
		},
		{
			key: { text: 'Email' },
			value: { text: email || 'Not provided' }
		},
		{
			key: { text: 'Address' },
			value: hasAddress
				? { html: addressToMultilineStringHtml(address) }
				: { text: 'Not provided' },
			actions: {
				items: [
					{
						text: hasAddress ? 'Change' : 'Add',
						href: `/appeals-service/appeal-details/${appealId}/interested-party-comments/${
							comment.id
						}/edit/address?review=${isReviewPage}&editAddress=${hasAddress ? 'true' : 'false'}`,
						visuallyHiddenText: 'address'
					}
				]
			}
		},
		...(isReviewPage
			? []
			: [
					{
						key: { text: 'Site visit requested' },
						value: { text: comment.siteVisitRequested ? 'Yes' : 'No' },
						actions: {
							items: comment.siteVisitRequested
								? [
										{
											text: 'Change',
											href: `/appeals-service/appeal-details/${appealId}/interested-party-comments/${comment.id}/edit/site-visit-requested`,
											visuallyHiddenText: 'site visited requested'
										}
								  ]
								: []
						}
					}
			  ]),
		{
			key: { text: 'When did the interested party submit the comment?' },
			value: { html: dateISOStringToDisplayDate(comment.created) }
		},
		{
			key: {
				text: comment.redactedRepresentation && redactMatching ? 'Original comment' : 'Comment'
			},
			value: {
				text: commentIsDocument ? 'Added as a document' : comment.originalRepresentation
			},
			actions: {
				items:
					commentIsDocument || isReviewPage || redactMatching
						? []
						: [
								{
									text: 'Redact',
									href: `/appeals-service/appeal-details/${appealId}/interested-party-comments/${comment.id}/redact`
								}
						  ]
			}
		},
		...(comment.redactedRepresentation && redactMatching
			? [
					{
						key: { text: 'Redacted comment' },
						value: { text: comment.redactedRepresentation },
						actions: {
							items: [
								{
									text: 'Change',
									href: `/appeals-service/appeal-details/${appealId}/interested-party-comments/${comment.id}/redact`
								}
							]
						}
					}
			  ]
			: []),
		{
			key: { text: 'Supporting documents' },
			value: attachmentsList ? { html: attachmentsList } : { text: 'No documents' },
			actions: {
				items: [
					...(filteredAttachments?.length
						? [
								{
									text: 'Manage',
									href: `/appeals-service/appeal-details/${appealId}/interested-party-comments/${comment.id}/manage-documents/${folderId}`,
									visuallyHiddenText: 'supporting documents'
								}
						  ]
						: []),
					{
						text: 'Add',
						href: `/appeals-service/appeal-details/${appealId}/interested-party-comments/${comment.id}/add-document`,
						visuallyHiddenText: 'supporting documents'
					}
				]
			}
		},
		...(isReviewPage || comment.status !== COMMENT_STATUS.INVALID
			? []
			: [
					{
						key: { text: 'Why was the comment rejected?' },
						value: { html: generateRejectionReasonsList(comment) }
					}
			  ])
	];

	return {
		type: 'summary-list',
		wrapperHtml: {
			opening: '<div class="govuk-grid-row"><div class="govuk-grid-column-full">',
			closing: '</div></div>'
		},
		parameters: { rows }
	};
}
