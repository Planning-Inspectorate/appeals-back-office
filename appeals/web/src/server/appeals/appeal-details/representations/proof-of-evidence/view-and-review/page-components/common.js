import { buildHtmlList } from '#lib/nunjucks-template-builders/tag-builders.js';
import { preRenderPageComponents } from '#lib/nunjucks-template-builders/page-component-rendering.js';
import { mapDocumentDownloadUrl } from '#appeals/appeal-documents/appeal-documents.mapper.js';
import { APPEAL_REPRESENTATION_TYPE } from '@pins/appeals/constants/common.js';

/** @typedef {import('#appeals/appeal-details/representations/types.js').Representation} Representation */

/**
 * Maps representation types to their respective URL path fragments.
 * @param {string} representationType
 * @returns {string}
 */
function mapRepresentationTypeToPath(representationType) {
	switch (representationType) {
		case APPEAL_REPRESENTATION_TYPE.APPELLANT_PROOFS_EVIDENCE:
			return 'appellant';
		case APPEAL_REPRESENTATION_TYPE.LPA_PROOFS_EVIDENCE:
			return 'lpa';
		default:
			throw new Error(`Unsupported representation type: ${representationType}`);
	}
}

/**
 * Generates the proof of evidence summary list used in both view and review pages.
 * @param {number} appealId
 * @param {Representation} proofOfEvidence - The representation object.
 * @returns {PageComponent} The generated proof of evidence summary list component.
 */
export function generateProofOfEvidenceSummaryList(appealId, proofOfEvidence) {
	const filteredAttachments = proofOfEvidence.attachments?.filter((attachment) => {
		const { isDeleted, latestVersionId } = attachment?.documentVersion?.document ?? {};
		return latestVersionId === attachment.version && !isDeleted;
	});

	const attachmentsList = filteredAttachments?.length
		? buildHtmlList({
				items: filteredAttachments.map(
					(a) =>
						`<a class="govuk-link" href="${mapDocumentDownloadUrl(
							a.documentVersion.document.caseId,
							a.documentVersion.document.guid,
							a.documentVersion.document.name
						)}" target="_blank">${a.documentVersion.document.name}</a>`
				),
				isOrderedList: true,
				isNumberedList: filteredAttachments.length > 1
		  })
		: null;

	const proofOfEvidenceTypePath = mapRepresentationTypeToPath(proofOfEvidence.representationType);
	const rows = [
		{
			key: { text: 'Proof of evidence and witnesses' },
			value: attachmentsList ? { html: attachmentsList } : { text: 'No documents' },
			actions: {
				items: [
					{
						text: 'Change',
						href: `/appeals-service/appeal-details/${appealId}/proof-of-evidence/${proofOfEvidenceTypePath}/change/${proofOfEvidence.attachments?.[0]?.documentVersion?.document?.folderId}`,
						visuallyHiddenText: 'proof of evidence and witnesses'
					}
				]
			}
		}
	];

	/** @type {PageComponent} */
	const summaryList = {
		type: 'summary-list',
		wrapperHtml: {
			opening: '<div class="govuk-grid-row"><div class="govuk-grid-column-full">',
			closing: '</div></div>'
		},
		parameters: { rows }
	};

	preRenderPageComponents([summaryList]);

	return summaryList;
}
