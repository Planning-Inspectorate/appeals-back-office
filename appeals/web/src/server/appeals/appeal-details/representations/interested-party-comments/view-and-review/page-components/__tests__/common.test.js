import { generateCommentSummaryList } from '#appeals/appeal-details/representations/interested-party-comments/view-and-review/page-components/common.js';

describe('generateCommentSummaryList', () => {
	it('should return the default comment summary list', () => {
		const appealId = 5;
		const comment = {};
		const filter = {};
		// @ts-ignore
		const result = generateCommentSummaryList(appealId, comment, filter);
		expect(result.parameters.rows[0].value.text).toContain('Not provided');
		expect(result.parameters.rows[1].value.text).toContain('Not provided');
		expect(result).toMatchSnapshot();
	});

	it('should return the comment summary list including data', () => {
		const appealId = 5;
		const comment = {
			represented: {
				address: {
					addressLine1: '123 Random street',
					postCode: 'SW1A 1AA'
				},
				name: 'John Doe',
				email: 'john.dow@example.com'
			},
			siteVisitRequested: true,
			whenSubmitted: '2021-01-01T00:00:00.000Z',
			originalRepresentation: 'Test comment'
		};
		const filter = {};
		// @ts-ignore
		const result = generateCommentSummaryList(appealId, comment, filter);
		expect(result.parameters.rows[0].value.text).toContain('John Doe');
		expect(result.parameters.rows[1].value.text).toContain('john.dow@example.com');
		expect(result).toMatchSnapshot();
	});
});
