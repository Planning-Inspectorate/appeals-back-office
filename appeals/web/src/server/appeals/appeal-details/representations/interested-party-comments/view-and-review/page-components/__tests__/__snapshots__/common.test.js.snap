// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateCommentSummaryList should return the comment summary list including data 1`] = `
{
  "parameters": {
    "rows": [
      {
        "key": {
          "text": "Interested party",
        },
        "value": {
          "text": "John Doe",
        },
      },
      {
        "key": {
          "text": "Email",
        },
        "value": {
          "text": "john.dow@example.com",
        },
      },
      {
        "actions": {
          "items": [
            {
              "href": "/appeals-service/appeal-details/5/interested-party-comments/undefined/edit/address?review=undefined&editAddress=true",
              "text": "Change",
              "visuallyHiddenText": "address",
            },
          ],
        },
        "key": {
          "text": "Address",
        },
        "value": {
          "html": "123 Random street, </br>SW1A 1AA",
        },
      },
      {
        "actions": {
          "items": [
            {
              "href": "/appeals-service/appeal-details/5/interested-party-comments/undefined/edit/site-visit-requested",
              "text": "Change",
              "visuallyHiddenText": "site visited requested",
            },
          ],
        },
        "key": {
          "text": "Site visit requested",
        },
        "value": {
          "text": "Yes",
        },
      },
      {
        "key": {
          "text": "When did the interested party submit the comment?",
        },
        "value": {
          "html": "",
        },
      },
      {
        "actions": {
          "items": [
            {
              "href": "/appeals-service/appeal-details/5/interested-party-comments/undefined/redact",
              "text": "Redact",
            },
          ],
        },
        "key": {
          "text": "Comment",
        },
        "value": {
          "text": "Test comment",
        },
      },
      {
        "actions": {
          "items": [
            {
              "href": "/appeals-service/appeal-details/5/interested-party-comments/undefined/add-document",
              "text": "Add",
              "visuallyHiddenText": "supporting documents",
            },
          ],
        },
        "key": {
          "text": "Supporting documents",
        },
        "value": {
          "text": "Not provided",
        },
      },
    ],
  },
  "type": "summary-list",
  "wrapperHtml": {
    "closing": "</div></div>",
    "opening": "<div class="govuk-grid-row"><div class="govuk-grid-column-full">",
  },
}
`;

exports[`generateCommentSummaryList should return the default comment summary list 1`] = `
{
  "parameters": {
    "rows": [
      {
        "key": {
          "text": "Interested party",
        },
        "value": {
          "text": "Not provided",
        },
      },
      {
        "key": {
          "text": "Email",
        },
        "value": {
          "text": "Not provided",
        },
      },
      {
        "actions": {
          "items": [
            {
              "href": "/appeals-service/appeal-details/5/interested-party-comments/undefined/edit/address?review=undefined&editAddress=false",
              "text": "Add",
              "visuallyHiddenText": "address",
            },
          ],
        },
        "key": {
          "text": "Address",
        },
        "value": {
          "text": "Not provided",
        },
      },
      {
        "actions": {
          "items": [],
        },
        "key": {
          "text": "Site visit requested",
        },
        "value": {
          "text": "No",
        },
      },
      {
        "key": {
          "text": "When did the interested party submit the comment?",
        },
        "value": {
          "html": "",
        },
      },
      {
        "actions": {
          "items": [
            {
              "href": "/appeals-service/appeal-details/5/interested-party-comments/undefined/redact",
              "text": "Redact",
            },
          ],
        },
        "key": {
          "text": "Comment",
        },
        "value": {
          "text": undefined,
        },
      },
      {
        "actions": {
          "items": [
            {
              "href": "/appeals-service/appeal-details/5/interested-party-comments/undefined/add-document",
              "text": "Add",
              "visuallyHiddenText": "supporting documents",
            },
          ],
        },
        "key": {
          "text": "Supporting documents",
        },
        "value": {
          "text": "Not provided",
        },
      },
    ],
  },
  "type": "summary-list",
  "wrapperHtml": {
    "closing": "</div></div>",
    "opening": "<div class="govuk-grid-row"><div class="govuk-grid-column-full">",
  },
}
`;
